gustaf/io/hmascii.py

```diff
@@ -0,0 +1,313 @@
+        --------
+        None
+        """
+        self.name = str(line.values[1])
+        self.elements = dict()  # element type: list()
+
+    def add_element(self, line):
+        """Parse an element line from an HMASCII file.
+
+        Parameters
+        -----------
+        line: HMLine
+
+        Returns
+        --------
+        None
+        """
+        element_type = line.name
+        if element_type not in self.elements:
+            self.elements[element_type] = list()
+        self.elements[element_type].append(
+                [
+                        int(node) for node in
+                        line.values[2:2 + self.element_types[element_type].
+                                    number_of_nodes]
+                ]
+        )
+
+
+class HMModel:
+    """Create a representation of a HyperMesh model from the HMASCII file
+    specified by `filename`.
+
+    Parameters
+    -----------
+    node_ids: dict
+    node_coordinates: list
+    components: list
+
+    Returns
+    --------
+    None
+    """
+
+    __slots__ = ["node_ids", "node_coordinates", "components"]
+
+    def __init__(self, filename):
+        """
+        Parameters
+        -----------
+        filename: str
+
+        Returns
+        --------
+        None
+        """
+        # read file into arrays
+        self.node_ids = dict()  # HM node id: internal node id
+        self.node_coordinates = list()
+        self.components = list()  # HMComponent objects
+
+        current_component = None
+
+        with open(filename, 'r') as hm_file:
+            for line_string in hm_file:
+                line = HMLine(line_string)
+
+                # read node
+                if line.name == "node":
+                    self.node_ids[int(line.values[0])] =\
+                        len(self.node_coordinates)
+                    self.node_coordinates.append(
+                            [float(coord) for coord in line.values[1:4]]
+                    )
+
+                # read component
+                elif line.name == "component":
+                    current_component = HMComponent(line)
+                    self.components.append(current_component)
+
+                # read element
+                elif line.name in HMComponent.element_types:
+                    if not current_component:
+                        raise RuntimeError(
+                                'Encountered element before first '
+                                'component.'
+                        )
+                    current_component.add_element(line)
+
+
+def load(fname, element_type=''):
+    """hmascii load.
+
+    Parameters
+    -----------
+    fname: str
+        The filename ending in '.hmascii'.
+    element_type: str
+        The volume element type to be imported. If this is not specified, the
+        most common element type in the main component is chosen.
+
+    Returns
+    -----------
+    mesh: Mesh
+    """
+
+    if not fname.endswith(".hmascii"):
```

Comment:
Here this only really works if the name element is a pathlib/etc object and not with strings. Could you either make a type check or convert it if it is not of this type? 
pathlib does not care if a pathlib object is used as the init value meaning that 
```
fname = pathlib.Path(fname)
```

should work if fname is a pathlib.Path object and a string.